src/plugins/shared_ux/public/components/redirect_app_links/redirect_app_links.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FunctionComponent, useRef, useMemo } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { ApplicationStart } from 'src/core/public';
+import { createNavigateToUrlClickHandler } from './click_handler';
+
+export interface RedirectAppLinksProps extends React.HTMLAttributes<HTMLDivElement> {
+  navigateToUrl: ApplicationStart['navigateToUrl'];
+  currentAppId$: ApplicationStart['currentAppId$'];
+  className?: string;
+  'data-test-subj'?: string;
+}
+
+/**
+ * Utility component that will intercept click events on children anchor (`<a>`) elements to call
+ * `application.navigateToUrl` with the link's href. This will trigger SPA friendly navigation
+ * when the link points to a valid Kibana app.
+ *
+ * @example
+ * ```tsx
+ * <RedirectAppLinks application={application}>
```

Comment:
Needs updating

Reply:
General question, maybe for @clintandrewhall , but should all component comments like this that kind of act as documentation to the `.mdx` file? There is some nicety to having it as a JSDoc comment so it appears in dev's IDE's directly. Maybe just a more succinct version in the component file and the more verbose in the `.mdx`?

Reply:
@cchaos Publicly-exposed constructs require comments like this for the auto-generated API docs.

https://github.com/elastic/kibana/blob/main/api_docs/shared_u_x.devdocs.json
https://github.com/elastic/kibana/blob/main/api_docs/shared_u_x.mdx

```
Public APIs missing comments
Total count of every public API that lacks a comment. Target amount is 0. Run node scripts/build_api_docs --plugin [yourplugin] --stats comments for more detailed information.
```

Reply:
These comments still need updating as the example still shows needing `application`/